client/state/terms/actions.js

```diff
@@ -84,6 +86,12 @@ export function updateTerm( siteId, taxonomy, termId, termSlug, term ) {
 					} ) );
 				} );
 
+				// Update the default category if needed
+				const siteSettings = getSiteSettings( state, siteId );
+				if ( taxonomy === 'category' && get( siteSettings, [ 'default_category' ] ) === termId && updatedTerm.ID !== termId ) {
```

Comment:
This conditional is quite long and having the `get` nested in there makes it hard to read at a glance, should we put each condition on its own line?